learn/sherif/main_test/src/main/java/frc/robot/RobotContainer.java

```diff
@@ -26,26 +28,37 @@
  */
 public class RobotContainer {
     // The robot's subsystems and commands are defined here...
-    private Factory m_f = new Factory();
-    private DriveTrain m_drivetrain;
-    private Joystick m_controller;
+    public Factory factory = new Factory();
+    private DriveTrain m_drivetrain = null;
+    private Joystick m_controller = null;
+    private IntakeControl m_intake = null;
 
     /**
      * The container for the robot.  Contains subsystems, OI devices, and commands.
      */
     public RobotContainer() {
         // Configure the button bindings
+        this.m_drivetrain = new DriveTrain(factory);
+        this.m_controller = new Joystick(0);
         configureButtonBindings();
     }
 
     // Allows for a (mock) controller to be injected
-    public RobotContainer(Joystick controller, DriveTrain train) {
-        this.m_controller = controller;
-        this.m_drivetrain = train;
+    public RobotContainer(RobotInjection injection) {
+        if (injection.joystick != null) {
+            this.m_controller = injection.joystick;
+        }
+        if (injection.driveTrain != null) {
+            this.m_drivetrain = injection.driveTrain;
```

Comment:
Make these name consistent.  Code should not refer to m_drivetrain on one object and driveTrain on another object.  If you are using `m_` prefix, then use it every where.  Similarly be consistent with name case.